File: tests/appium/config/appium_config.py
import os
from pathlib import Path

class AppiumConfig:
    APPIUM_SERVER_URL = os.environ.get("APPIUM_SERVER_URL", "http://127.0.0.1:4723")
    PLATFORM_NAME = "Android"
    AUTOMATION_NAME = "UiAutomator2"
    DEVICE_NAME = os.environ.get("ANDROID_DEVICE_NAME", "Android Emulator")
    APP_PACKAGE = "com.cephgrow.ai"
    APP_ACTIVITY = ".MainActivity"
    
    PROJECT_ROOT = Path(__file__).resolve().parent.parent.parent.parent
    APK_PATH = os.environ.get(
        "APK_PATH",
        str(PROJECT_ROOT / "frontend" / "android" / "app" / "build" / "outputs" / "apk" / "debug" / "app-debug.apk")
    )
    
    IMPLICIT_WAIT = int(os.environ.get("APPIUM_IMPLICIT_WAIT", "15"))
    EXPLICIT_WAIT = int(os.environ.get("APPIUM_EXPLICIT_WAIT", "20"))
    
    @classmethod
    def get_capabilities(cls) -> dict:
        caps = {
            "platformName": cls.PLATFORM_NAME,
            "appium:automationName": cls.AUTOMATION_NAME,
            "appium:deviceName": cls.DEVICE_NAME,
            "appium:appPackage": cls.APP_PACKAGE,
            "appium:appActivity": cls.APP_ACTIVITY,
            "appium:noReset": False,
            "appium:fullReset": False,
            "appium:newCommandTimeout": 180,
            "appium:autoGrantPermissions": True
        }
        if os.path.exists(cls.APK_PATH):
            caps["appium:app"] = cls.APK_PATH
        return caps

File: node_testing/appium_mobile/appiumConfig.js
class AppiumNodeConfig {
    static APPIUM_HOST = process.env.APPIUM_HOST || '127.0.0.1';
    static APPIUM_PORT = parseInt(process.env.APPIUM_PORT || '4723');
    static PLATFORM_NAME = 'Android';
    static AUTOMATION_NAME = 'UiAutomator2';
    static DEVICE_NAME = process.env.ANDROID_DEVICE_NAME || 'Android Emulator';
    static APP_PACKAGE = 'com.cephgrow.ai';
    static APP_ACTIVITY = '.MainActivity';
    
    static getCapabilities() {
        return {
            platformName: this.PLATFORM_NAME,
            'appium:automationName': this.AUTOMATION_NAME,
            'appium:deviceName': this.DEVICE_NAME,
            'appium:appPackage': this.APP_PACKAGE,
            'appium:appActivity': this.APP_ACTIVITY,
            'appium:noReset': true,
            'appium:newCommandTimeout': 120
        };
    }
}

module.exports = AppiumNodeConfig;
